packages/siopv2-oid4vp-op-auth/src/services/IdentifierService.ts

```diff
@@ -0,0 +1,77 @@
+import { KeyUse } from '@sphereon/ssi-sdk-ext.did-resolver-jwk'
+import { _ExtendedIKey } from '@veramo/utils'
+import {
+  CreateIdentifierArgs,
+  DID_PREFIX,
+  GetAuthenticationKeyArgs,
+  GetIdentifierArgs,
+  GetOrCreatePrimaryIdentifierArgs,
+  IdentifierAliasEnum,
+  IdentifierOpts,
+  KeyManagementSystemEnum,
+  SupportedDidMethodEnum,
+} from '../types/identifier'
+import { IDIDManager, IIdentifier, IKey, IResolver, TAgent } from '@veramo/core'
+import { getFirstKeyWithRelation } from '@sphereon/ssi-sdk-ext.did-utils'
+import { Siopv2HolderEvent } from '../types'
+
+export const getIdentifier = async (args: GetIdentifierArgs): Promise<IdentifierOpts> => {
+  const { keyOpts, context } = args
+
+  const identifier =
+    keyOpts.identifier ??
+    (await getOrCreatePrimaryIdentifier({
+      context,
+      opts: {
+        method: keyOpts.didMethod,
+        createOpts: { options: { type: keyOpts.keyType, use: KeyUse.Signature, codecName: keyOpts.codecName } },
+      },
+    }))
+  const key: _ExtendedIKey = await getAuthenticationKey({ identifier, context })
+  const kid: string = key.meta.verificationMethod.id
+
+  return { identifier, key, kid }
+}
+
+export const getAuthenticationKey = async (args: GetAuthenticationKeyArgs): Promise<_ExtendedIKey> => {
```

Comment:
A better version of this method lives in oid4vci-holder. Probably these should be moved to ssi-sdk-ext.did-utls